### DISCUSSION TITLE
A little more handholding, please

### DISCUSSION THREAD

**Discussion by tuuktuuk** [endless-sky/endless-sky#10878] [Balance, Mechanics, and Interactions]
So, here's something after 200 hours of playing this game. I prefer trying to always gain all the information in-game and not looking for it in Wikis. There were two big problems I had, both of them turning out to be user interface issues.

So, the two things:
1) I had played the game for about 20 hours when I at last found out that it really has a single consistent plot. Yeah, it was fun helping the viking guys and all, but I would have preferred the game doing a bit more to push me towards actually finding the main missions. The main problems were that I did not even know the game has a main plot and that I hadn't understood that every time I leave a planet, I should first press the button called "Spaceport". Now it's a reflex and all's fine. But, there should be something to motivate the player to always check the spaceport. Either the Spaceport button should do something useful a LOT more often, or there should be something that tells more explicitly about the importance of clicking it every time you're on a planet! And if you need to press it every time, it's a bit questionable whether there should even be a separate Spaceport button...
Of course the game could also ask "do you want to leave the planet although you haven't checked the spaceport yet?" ...at least in the Free Worlds missions. Also, the Free Worlds missions could kind of come catch me from wherever I am and railroad me into getting to the planet where they will continue. 

2) I got stuck with getting the control systems of the two subspecies' drones, because I didn't figure out that I can actually pick individual components off ships I had not taken over. I tried capturing the ships and bringing them to some place with an outfitter to strip the components away from the ship. But, these ships just explode EVERY time I try that, so in concluded that this is something I cannot do for now and need to find some place somewhere that gives me the ability to capture these ships. I figured it must have something to do with the control room, but I could not understand how.
Here the birds could just give me a hint. If the player tries capturing five of those drones, the next time there could be something like
– "I don't seem to be able to capture the drones. They just explode immediately!"
– "We do not need the whole ship, only the control system. Strip them off the ship in space, right where the ship has been disabled and then let the dismantled ship stay floating there. Make sure you have space on your ship and haven't filled the cargo space with trade goods or outfit modules!"
– "Oh, so it is unnecessary trying to capture these drones in the first place, just boarding them is enough?"
– "Correct. Now good luck getting the drones. We need them immediately. Your ship is properly equipped for stripping the control module off the drones, so make haste!"

This kind of dialogue might be necessary because the current mechanics of stripping components off ships is very unintuitive: Why can I get a jump drive off a ship without fighting its crew?! If everything is located outside the hull, why does the lore explain the reason for this? I had played for about 180 hours when I noticed that the dialogue where I usually click "Attempt capture" and that shows the components of the ship actually allows me to "Take" something from it. The main reason for the confusion was that the "Take" button was always grayed out, as I was always carrying the maximum amount of trade goods with me. One time I accidentally forgot to load goods on my ship before departing a planet, and happened to notice the component list being of different colour and tried out some things. It would feel much more logical that you can only disassemble a ship at an outfitter. Or, it should be explained by the game why all tiniest pirate ships have the tools for carefully disassembling other ships in middle of the space.

**Comment by Hecter94** [comment]
Hey! Thanks for the feedback, there's quite a bit here so I'm just going to go through a couple of the things you brought up to make sure we're on the same page.

> But, there should be something to motivate the player to always check the spaceport. Either the Spaceport button should do something useful a LOT more often, or there should be something that tells more explicitly about the importance of clicking it every time you're on a planet! 

There are a couple things that point you towards the spaceport, 
At the start of the game, you're given the following tutorial dialogue: `Taking on jobs is a safe way to earn money. Special missions are offered in the Spaceport; more mundane jobs are posted on the Job Board. Most special missions are only offered once: if you turn one down, it will not be offered to you again.`

When you finish the tutorial with James, he tells you: `"Best of luck to you, Captain. Don't hesitate to look me up if you're ever in the area. And remember to check the spaceports wherever you land. You never know who you'll find wandering around there."`

If you do check the spaceport immediately after James tells you to, we reinforce that behavior by ensuring you'll get a mission: https://github.com/endless-sky/endless-sky/pull/8560

If the player still doesn't check spaceports, we've also added a mission that will pop up and specifically tell the player to check spaceports: https://github.com/endless-sky/endless-sky/pull/6595
`"You know, I'm a captain too. I know how quickly the luster of the galaxy can fade into the weekly grind if you wait for opportunity to come knocking. So why keep waiting? I'm sure you'll find something new to do in the spaceport."`

`Yeah, it can feel a little pointless running up against the same loud-mouthed people all the time. But just because a spaceport doesn't have anything to offer one day doesn't mean that it'll never have anything worthwhile. Wait a while, and there might be something new."`

Do you remember seeing the tutorial help text telling you that special missions are offered in the spaceport, the dialogue from James telling you to check spaceports, and the special mission that should have triggered, which also would have told you to check spaceports?

>    Also, the Free Worlds missions could kind of come catch me from wherever I am and railroad me into getting to the planet where they will continue.

We don't want to railroad the player into anything. A few things will push players towards the main campaign and a few different entry points. In the future, there will be more "main" campaigns to choose from, so we don't want to take away player agency and railroad them into doing things in the "correct" way in the "correct" order or at the "correct" time.

> I tried capturing the ships and bringing them to some place with an outfitter to strip the components away from the ship. But, these ships just explode EVERY time I try that, so in concluded that this is something I cannot do for now and need to find some place somewhere that gives me the ability to capture these ships. I figured it must have something to do with the control room, but I could not understand how.

`self-destruct` is a random chance that is rolled twice, once when you try to board the ship and once when you try to capture it. If you just want to loot outfits from the ship, you only need to successfully pass the self-destruct chance once; if you want to capture the ship, you have to pass the self-destruct chance twice. You don't need to do anything special to capture those ships beyond just passing the self-destruct chance a second time. 

> This kind of dialogue might be necessary because the current mechanics of stripping components off ships is very unintuitive

This is explained and tutorialized during the main plot, specifically FW Katya 3 in the Free Worlds prologue, where you're told to disable a surveillance drone, plunder its Surveillance Pod and return it to Katya. 
Do you remember doing that mission? I'm fairly sure that you would have needed to complete that mission and thus be introduced to boarding/looting in order to be offered the Wanderer mission to board/loot the Korath Automata. 
If FW Katya 3 isn't clear enough or doesn't teach the mechanic well enough, we could perhaps tweak the wording of that mission a bit to make it a bit more explicit to the player exactly what they need to do.

**Reply by ziproot** [reply]
Addendums:

> This is explained and tutorialized during the main plot, specifically FW Katya 3 in the Free Worlds prologue, where you're told to disable a surveillance drone, plunder its Surveillance Pod and return it to Katya.

You actually are given two opportunities to do this, as you also steal an electron beam from a gunboat, and in checkmate, you are given three opportunities since you steal a Jump Drive from the Pug, though I assume most people still choose reconciliation.

The thesis statement is very similar to an already existing issue: #3663 

> Also, the Free Worlds missions could kind of come catch me from wherever I am and railroad me into getting to the planet where they will continue.

This is an issue the developers are aware about, but once the Syndicate and Republic campaigns (and possibly Pirate and Merchant campaigns) become available, it will be much harder to avoid all of them. Right now there are some missions that lead you to Free Worlds space, but they offer before the Free Worlds actually exists. A solution could be some (semi)-lucrative jobs to and from Free Worlds space.

**Reply by tuuktuuk** [reply]
> > But, there should be something to motivate the player to always check the spaceport. Either the Spaceport button should do something useful a LOT more often, or there should be something that tells more explicitly about the importance of clicking it every time you're on a planet!
> 
> There are a couple things that point you towards the spaceport, At the start of the game, you're given the following tutorial dialogue: `Taking on jobs is a safe way to earn money. Special missions are offered in the Spaceport; more mundane jobs are posted on the Job Board. Most special missions are only offered once: if you turn one down, it will not be offered to you again.`
> 
> When you finish the tutorial with James, he tells you: `"Best of luck to you, Captain. Don't hesitate to look me up if you're ever in the area. And remember to check the spaceports wherever you land. You never know who you'll find wandering around there."`
> 
> If you do check the spaceport immediately after James tells you to, we reinforce that behavior by ensuring you'll get a mission: #8560

Sounds cool :)
There's always so much a tutorial is telling you about that a lot of things will end up not being remembered. In this case, I think, I have dismissed James's words as just NPC chatter. Maybe because I wasn't familiar with the UI yet and saying "remember to check the spaceports whenever you land" didn't directly connect with anything I had already experienced.

But yeah, it's a bit difficult fixing stupid ;) If I don't manage to remember things told to me by the tutorial, there's very little the developers can do about it.
The only useful thing I can suggest here is checking how often the player presses the Spaceport button during the, say, first 20 landings and/or 50 landings. Especially when told after a really large number of landings, the information might be more able to push through into even the densest of brains :D I mean, if after 100 landings the game checks what I've done in the last 20 landings and I have only pressed the "spaceport" button a maximum of 5 times out of those 20, something is probably wrong in my head and I should be healed, possibly by braking the 4th wall.
Also, maybe have a picture of a spaceship's driver's desk with buttons in a very similar looking layout to how it looks when you've landed on a planet, could help the player understand what they are supposed to be doing ;)
> 
> If the player still doesn't check spaceports, we've also added a mission that will pop up and specifically tell the player to check spaceports: #6595 `"You know, I'm a captain too. I know how quickly the luster of the galaxy can fade into the weekly grind if you wait for opportunity to come knocking. So why keep waiting? I'm sure you'll find something new to do in the spaceport."`
> 
> `Yeah, it can feel a little pointless running up against the same loud-mouthed people all the time. But just because a spaceport doesn't have anything to offer one day doesn't mean that it'll never have anything worthwhile. Wait a while, and there might be something new."`
> 

This is awesome :) I think I ran into this one, and it did increase my [Spaceport]-clicking, but not enough to make it into a habit :))

> We don't want to railroad the player into anything. A few things will push players towards the main campaign and a few different entry points. In the future, there will be more "main" campaigns to choose from, so we don't want to take away player agency and railroad them into doing things in the "correct" way in the "correct" order or at the "correct" time.
> 

Yeah, I think this problem will probably disappear by time, as this awesome thing gets closer to being a ready product :) 

> 
> `self-destruct` is a random chance that is rolled twice, once when you try to board the ship and once when you try to capture it. If you just want to loot outfits from the ship, you only need to successfully pass the self-destruct chance once; if you want to capture the ship, you have to pass the self-destruct chance twice. You don't need to do anything special to capture those ships beyond just passing the self-destruct chance a second time.

Hah, then I just had an extremely bad luck with the RNG. I think I tried capturing about 30 drones before "understanding" that it is not possible at all in the game. 

> This is explained and tutorialized during the main plot, specifically FW Katya 3 in the Free Worlds prologue, where you're told to disable a surveillance drone, plunder its Surveillance Pod and return it to Katya. Do you remember doing that mission? I'm fairly sure that you would have needed to complete that mission and thus be introduced to boarding/looting in order to be offered the Wanderer mission to board/loot the Korath Automata. If FW Katya 3 isn't clear enough or doesn't teach the mechanic well enough, we could perhaps tweak the wording of that mission a bit to make it a bit more explicit to the player exactly what they need to do.

What happened to me here is that I captured the surveillance drone, went to an outfitter and removed the surveillance pod there. I did the same with the pug ships – captured four of them, flew them to an outfitter, took the weapons and jump drives I needed. And then four of my ships had jump drives (and the other ~30 ships did not).
What happened was that this part of the tutorial actually reinforced my misunderstanding that outfits are removed from the ships at an outfitter. One solution could be to grey out the "Attempt capture" button when I'm boarding a surveillance drone. That way I won't be able to misunderstand what looting a ship outfit actually means. Though, at that point I had already captured about 15 pirate ships for my fleet, so just graying out the "Attempt capture" might not have been enough for helping this oblivious player understand that capturing is not what I should be looking into. Especially since I know the game is still work in progress.

Anyway, I absolutely love the plot of this game! I got so giddy from creating a new species yesterday – was wondering if I've ever player a game with such a well-written plot! Planescape:Torment maybe, but that's pretty much all the competition there is. Keep up the goo^H^H^Hawesome job! <3
Now I'm off for my next 200 hours of playing this, and then, as the game is getting better and better, probably for the remaining 400, getting ES past Factorio ;)

![зображення](https://github.com/user-attachments/assets/6f0bfa95-0a6a-46a8-b872-816dc152b0cf)

**Reply by slchorne** [reply]
FW Katya 3 doesn't tell you **how** to do it, if you accept the mission the dialog is

`"Great!" says Katya. "All we have to do is disable a drone, board it, yank out the surveillance equipment, and bring it back here to test it out."`

As @tuuktuuk experienced, their cargo was full so they never had an opportunity to learn what that the "Take" button did.

And if you come back empty handed the dialog is:

`You land on <planet>, but you don't have a Surveillance Pod. You must disable a Navy surveillance drone and plunder the outfit from it before returning`

**Comment by Quotatron3000** [comment]
>  I have dismissed James's words as just NPC chatter.

When I started out on Endless Sky I had the exact same reaction. That is because the Spaceport spams "flavour text" that just looks the kind of background NPC dialogue most players will phase out psychologically. Not all planets have an interesting Spaceport event, and new players will try it on planets with a low probability of getting a unique event.

Players are efficient, and will not waste time on an interaction that does not show evident signs of being _consistently_ beneficial. Spaceport just seems like a 'spam flavour text' button, the kind of background noise devs add to NPCs who say things but don't actually benefit the player.

From a practical standpoint, I would propose either completely removing NPC flavour text, or reworking it so the flavour text acts as helpful hints to the player. Examples like:
'Wow, I can't believe you can buy so much Food in the Trading section! They always want food on [planet with a good price on food]!'
'There's no work opportunities in this spaceport due to the economy, but I bet there's work on opportunities on [planet name with a spaceport work opportunity]!'
'[Planet] is looking for [workers]'

Mashing the spaceport button enough times (about 5 to 10 times) should also auto-generate a passenger job that leads the player to a planet with an "actual" spaceport opportunity.

**Add a: 'How Do I...?' button**

One thing that has frustrated me is there are licences required to get things, but I have no idea how I get these licences. I'd like to be able to ask the game 'How do I get a republic licence for a warship?' or 'How do I get a remnant licence to buy weapons?'. Pressing the button gives contextual options. Like if I'm on a Republic planet it might offer:
'How Do I...
...Join the Navy?
...Get a Republic licence?
...Purchase a Republic ship?
...Trade?
...Board a ship?'
etc

**Comment by TobiMcNamobi** [comment]
I'm like 5 hours in the game. And now I read about the importance of spaceports. I agree they could be highlighted better indeed. Maybe they could have a special frame or a sticker or some other highlight if they contain a story mission.

**Comment by slchorne** [comment]
There was a feature request #7449 that addresses some of this. In the player information panel add a page that just says "X of N missions completed in storyline" and lists any other incomplete side quests, so you have a running tally of progress

I agree with the drone capturing problems, there needs to be a better fleshed out tutorial mission that walks a player through plundering and capturing other ships
